File: static/python/intersect_sorted_arrays1.py
# Intersect_sorted_arrays1.h 4a4c5f91493e5e482eaa79892816c1ccefa084f4
# @include
def intersect_two_sorted_arrays(A, B):
    intersection_A_B = []
    for i in range(len(A)):
        if i == 0 or A[i] != A[i - 1]:
            for b in B:
                if A[i] == b:
                    intersection_A_B.append(A[i])
                    break
    return intersection_A_B
# @exclude

File: solutions/java/IntersectSortedArrays1.java
import java.util.ArrayList;
import java.util.List;

public class IntersectSortedArrays1 {
  // @include
  public static List<Integer> intersectTwoSortedArrays(List<Integer> A,
                                                       List<Integer> B) {
    List<Integer> intersectionAB = new ArrayList<>();
    for (int i = 0; i < A.size(); ++i) {
      if ((i == 0 || A.get(i) != A.get(i - 1)) && B.contains(A.get(i))) {
        intersectionAB.add(A.get(i));
      }
    }
    return intersectionAB;
  }
  // @exclude
}
